fb-negotiation/core/controller.py

```diff
@@ -11,18 +11,25 @@ def event_callback(self, event):
         if event.action == 'select':
             self.marked_agree[event.agent] = True
             self.outcomes[event.agent] = event.data
+        elif event.action == 'reject':
+            self.quit = True
+            self.outcomes[event.agent] = event.data
         elif event.action == 'quit':
             self.quit = True
 
     def valid_end_state(self):
-        first_agent_proposal = self.outcomes[0]['item_split']
-        second_agent_proposal = self.outcomes[1]['item_split']
-        items = ['book', 'hat', 'ball']
+        try:
+            first_agent_proposal = self.outcomes[0]['item_split']
+            second_agent_proposal = self.outcomes[1]['item_split']
+        except TypeError:
+            self.outcomes[0] = {'item_split': 'deal_rejected'}
+            self.outcomes[1] = {'item_split': 'deal_rejected'}
+            return False
 
-        for item in items:
-            item_count = self.scenario.kbs[0].facts['Item_counts']
+        for item in ['book', 'hat', 'ball']:
+            item_count = self.scenario.kbs[0].facts['Item_counts'][item]
```

Comment:
Shall we use `for item, count in self.scenario.kbs[0].facts['Item_counts'].iteritems()` so that there's no hard-coded stuff?

Reply:
Sounds like you really plan to expand this in the future.  No problem, changed!